bears/vcs/git/GitCommitBear.py

```diff
@@ -203,3 +243,94 @@ def check_body(self, body,
+            of short references like ``closes #213``. Otherwise behaves
+            according to other chosen flags.
+            More on keywords follows.
+            [GitHub](https://help.github.com/articles/closing-issues-via-commit-messages/)
+            [GitLab](https://docs.gitlab.com/ce/user/project/issues/automatic_issue_closing.html)
+        :param body_close_issue_full_url:
+            Checks the presence of issue close reference with a full URL
+            related to some issue. Works along with ``body_close_issue``.
+        :param body_close_issue_on_last_line:
+            When enabled, checks for issue close reference presence on the
+            last line of the commit body. Works along with
+            ``body_close_issue``.
+        """
+        if not body_close_issue:
+            return
+
+        host = self.get_host_from_remotes()
+        if host not in self.SUPPORTED_HOST_KEYWORD_REGEX:
+            return
+
+        if body_close_issue_on_last_line:
+            body = body.splitlines()[-1]
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference in the last line.')
+        else:
+            result_message = ('Body of HEAD commit does not contain any {} '
+                              'reference.')
+
+        if body_close_issue_full_url:
+            result_info = 'full issue'
+            issue_ref_regex = (
+                r'https?://{}\S+/issues/(\w+)'.format(re.escape(host)))
+        else:
+            result_info = 'issue'
+            issue_ref_regex = r'#(\w+)'
+
+        concat_regex = '|'.join(kw for kw in self.CONCATENATION_KEYWORDS)
+        compiled_joint_regex = re.compile(
+            r'(?:{0})\s+'           # match issue related keywords,
+                                    # eg: fix, closes etc.
+
+            r'((?:\S(?!{1}))*\S'    # match links/tags
+                                    # eg: fix #123, fix https://github.com
+
+            r'(?:\s*(?:{1})\s*'     # match conjunctions like ',','and'
+
+            r'(?!{0})'              # reject if new keywords appear
+
+            r'(?:\S(?!{1}))*\S)*)'  # match links/tags followed after
+                                    # conjunctions if any
+            r''.format(
+                self.SUPPORTED_HOST_KEYWORD_REGEX[host],
+                concat_regex))
+
+        matches = compiled_joint_regex.findall(body)
+
+        if len(matches) == 0:
+            yield Result(self, result_message.format(result_info))
+            return
+
+        compiled_issue_ref_regex = re.compile(issue_ref_regex)
+        compiled_issue_no_regex = re.compile(r'[1-9][0-9]*')
+        compiled_concat_regex = re.compile(
+            r'\s*(?:{})\s*'.format(concat_regex))
+
+        for match in matches:
+            for issue in re.split(compiled_concat_regex, match):
+                reference = compiled_issue_ref_regex.match(issue)
```

Comment:
example when it won't work:
```
Fixes #123-4
```
(as `\w` doesn't match `-`. We should maybe use `\S`^^)